docs/core/tools/csproj.md

```diff
@@ -235,6 +235,12 @@ Specifies the URL for the repository where the source code for the package resid
 ### RepositoryType
 Specifies the type of the repository. Default is "git". 
 
+### RepositoryBranch
+Specifies the name of the source branch in the repository. When the project is packaged in a Nuget package it is added to the package metadata.
+
+### RepositoryCommit
+Specifies the commit hash for the source. When the project is packaged in a Nuget package this commit hash is added to the package metadata.
```

Comment:
"commit hash" is specific to git. Since RepositoryType may be different from git (e.g. tfvc) we shouldn't assert that this is always a commit hash. It could be a changeset number.

Reply:
Here's what the NuGet side says:

https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#pack-target


> Optional repository commit or changeset to indicate which source the package was built against. RepositoryUrl must also be specified for this property to be included. Example: 0e4d1b598f350b3dc675018d539114d1328189ef (NuGet 4.7.0+)
--